helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -92,19 +98,52 @@ public ZkBaseDataAccessor(HelixZkClient zkClient) {
     _zkClient = zkClient;
   }
 
+  /**
+   * Lazy initialization is used to reduce the cost of unnecessary opened zkConnections
+   * @return singleton instance of non-ZnRecord ZkClient
+   */
+  private HelixZkClient getNonZNRecordZkClient() {
+    if (_nonZNRecordZkClient == null) {
+      synchronized (this) {
+        if (_nonZNRecordZkClient == null) {
+          _nonZNRecordZkClient = new ZkClient.Builder().setConnection(
+              new ZkConnection(_zkClient.getServers(), ZkClient.DEFAULT_SESSION_TIMEOUT))
+              .setZkSerializer(new ZkSerializer() {
+                @Override
+                public byte[] serialize(Object data)
+                    throws ZkMarshallingError {
+                  if (data instanceof byte[]) {
+                    return (byte[]) data;
+                  }
+                  throw new HelixException("Only support a byte array as an argument!");
+                }
+
+                @Override
+                public Object deserialize(byte[] data)
+                    throws ZkMarshallingError {
+                  return data;
+                }
+              }).build();
+        }
+      }
+    }
+
+    return _nonZNRecordZkClient;
+  }
+
   /**
    * sync create
    */
   @Override
   public boolean create(String path, T record, int options) {
-    AccessResult result = doCreate(path, record, options);
+    AccessResult result = doCreate(_zkClient, path, record, options);
     return result._retCode == RetCode.OK;
   }
 
   /**
-   * sync create
+   * sync create. In order to support non-ZnRecord creation, the instance of {@link HelixZkClient} needs to be a parameter
    */
-  public AccessResult doCreate(String path, T record, int options) {
```

Comment:
We need to keep the old public API method signature. Otherwise, this diff will not be backward-compatible. Could you refactor so that we still have all the old public API methods that has the same behavior?

Reply:
For example, here you could do:

```
public AccessResult doCreate(String path, T record, int options) {
    doCreate(_zkClient, path, record, options);
}
```

Reply:
This is a really good idea, thanks!